src/http.ts

```diff
@@ -116,6 +116,7 @@ export class HttpClient<TContext = never> {
       authHeaders = authFactory ? await authFactory(requestContext as TContext) : {}
     } catch (error) {
       logger?.error('Authentication via authFactory failed', { error })
+      throw error
```

Comment:
Will this cause a 401/403 type HTTP code?

Reply:
No - the point of my change is to _prevent_ running the request if the auth factory function fails.

Previously the request would run anyway, without the result of the auth factory, and you'd end up with a confusing failed request which should never have started on top of the auth factory error.